src/inc/corprof.idl

```diff
@@ -2375,6 +2375,34 @@ interface ICorProfilerCallback7 : ICorProfilerCallback6
 }
 
 
+[
+    object,
+    uuid(5BED9B15-C079-4D47-BFE2-215A140C07E0),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback8 : ICorProfilerCallback7
+{
+    // This event is triggered whenever a metadata less method is jit compiled.
+    // These include IL Stubs, DynamicMethod or more generally any use of LCG.
+    // The goal is to provide profiler writers with enough information to identify
+    // it to users as beyond unknown code addresses.
+    // Note: FunctionID's provided here cannot be used to resolve to their metadata
+    //       tokens since LCG methods have no metadata.
+    HRESULT LightweightJITCompilationStarted(
+        FunctionID functionId,
+        UINT_PTR   ilHeader,
+        [in, string] const WCHAR * className,
+        [in, string] const WCHAR * methodName,
+        [in, string] const WCHAR * methodSignature);
+
```

Comment:
I'm fine with the seperate FunctionHasNoMetadata method that you suggest.

I think its reasonable if GetFunctionFromIP works for NGEN'ed methods too, but as mentioned in the other thread you won't be able to go the next step and resolve from FunctionID -> interesting info about the method until it has been restored.